pdns/dnsdist-lua.hh

```diff
@@ -129,14 +129,18 @@ private:
   uint32_t d_max{std::numeric_limits<uint32_t>::max()};
 };
 
-typedef boost::variant<string, vector<pair<int, string>>, std::shared_ptr<DNSRule>, DNSName, vector<pair<int, DNSName> > > luadnsrule_t;
+template <class T> using LuaTable = std::vector<std::pair<int, T>>;
+template <class T> using LuaAssociativeTable = std::unordered_map<std::string, T>;
+template <class T> using LuaTypeOrListOf = boost::variant<T, LuaTable<T>>;
```

Comment:
it might be clearer to rename `LuaTable` to `LuaList` or `LuaArray` - I like `LuaArray` (or vector?). Then, `LuaTypeOrListOf` should use the same word instead of `List`.

Reply:
`LuaArray` sounds better indeed, and then `LuaTypeOrListOf` becomes `LuaTypeOrArrayOf`. I'll change that!

Reply:
Done!